test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13018,11 +13018,21 @@ def __init__(self, username=None, attrs={}, jobname=None):
         self.interactive_handle = None
         if self.platform == 'cray' or self.platform == 'craysim':
             if 'Resource_List.select' in attrs:
-                select = attrs['Resource_List.select']
-                m = select.startswith('vntype=') or ':vntype=' in select
-                if not m:
-                    select = select + ":vntype=cray_compute"
-                    attrs['Resource_List.select'] = select
+                ra = []
+                r = attrs['Resource_List.select'].split('+')
+                for i in r:
+                    select = PbsTypeSelect(i)
+                    novntype = 'vntype' not in select.resources.keys()
+                    nohost = 'host' not in select.resources.keys()
+                    novnode = 'vnode' not in select.resources.keys()
```

Comment:
No need to say x in dict.keys().  You can directly say x in dict(e.g.,  'vntype' not in select.resources).  It's equivalent.  Same with the rest of the dictionaries.